app/models/spree/country.rb

```diff
@@ -6,7 +6,7 @@ class Country < ApplicationRecord
 
     validates :name, :iso_name, presence: true
 
-    def self.cached_find_by(attrs)
+    def self.find_by_cached(attrs)
```

Comment:
I looked at the Rails code to see what their method does differently to ours. They are actually not caching the result of the query. They are just caching the prepared statement for the query.

Rails actually defines `find_by_*` methods for every attribute. So there's `find_by_id` and `find_by_name`. If a model had the attribute `cached` then this method would collide again. Of course that's not going to happen with Spree::Country but someone may copy this code to another model.

Maybe we can rewrite the code so that we don't even need this method. I'll have a go at it.